src/main/python/covata/delta/utils.py

```diff
@@ -15,30 +15,52 @@
 import logging
 import inspect
 
-__all__ = ["LogMixin"]
 
-
-class LogMixin(object):
+class LogMixin:
     @property
     def logger(self):
-        return logging.getLogger(self.__caller())
-
-    def __caller(self):
-        """
-        Gets the name of the caller in {package}.{module}.{class} format
-
-        :return: the name of the caller
-        """
-        # type: () -> str
-        stack = inspect.stack()
-        if len(stack) < 3:
-            return ''
-
-        caller_frame = stack[2][0]
-        module = inspect.getmodule(caller_frame)
-        name = filter(lambda x: x is not None, [
-            module.__name__ if module else None,
-            self.__class__.__name__])
-
-        del caller_frame
-        return ".".join(name)
+        return logging.getLogger(caller())
+
+
+def caller():
+    """
+    Gets the name of the caller in {package}.{module}.{class} format
+
+    :return: the caller name
+    :rtype: str
+    """
+    stack = inspect.stack()
+    if len(stack) < 3:
+        return ''
+
+    caller_frame = stack[2][0]
+    module = inspect.getmodule(caller_frame)
+
+    name = filter(lambda x: x is not None, [
+        module.__name__ if module else None,
+        caller_frame.f_locals['self'].__class__.__name__
+        if 'self' in caller_frame.f_locals else None])
+    del caller_frame
+    return ".".join(name)
+
+
+def check_arguments(arguments, test_type, test_function, fail_message):
+    def decorator(function):
+        def _f(*args, **kwargs):
+            for arg, value in kwargs.items():
+                if arg in arguments:
+                    if not any(value is t or isinstance(value, t)
+                               for t in test_type):
+                        raise TypeError(
```

Comment:
not sure if raising `TypeError` is pythonic. Enforcing exact type (i.e. bear-typing) goes against duck-typing principles.